exporters/otlp/src/OpenTelemetry/Exporter/OTLP.hs

```diff
@@ -274,13 +279,17 @@ otlpExporter conf = do
                 threadDelay (retryDelay `shiftL` backoffCount)
                 sendReq req (backoffCount + 1)
 
-      either print (\_ -> pure ()) eResp
-
       case eResp of
-        Left err@(HttpExceptionRequest _ e) ->
-          if isRetryableException e
-            then exponentialBackoff
-            else pure $ Failure $ Just $ SomeException err
+        Left err@(HttpExceptionRequest req e)
+          | HTTPClient.host req == "localhost"
+          , HTTPClient.port req == 4317 || HTTPClient.port req == 4318
+          , ConnectionFailure _someExn <- e
+            -> do
+              pure $ Failure Nothing
```

Comment:
This works locally to swallow the exception. I still want to do this because otherwise the exponential backoff takes a very long time to wait.

I think we *probably* want to be a little more careful than this. But it's a good start.